src/python/common/path_pairs_config.py

```
@@ -0,0 +1,174 @@
+        with self._lock:
+            new_pairs = [p for p in self._pairs if p.id != pair_id]
+            if len(new_pairs) == len(self._pairs):
+                raise ValueError("PathPair with id '{}' not found".format(pair_id))
+            self._pairs = new_pairs
+
+    @classmethod
+    def from_str(cls, content: str) -> "PathPairsConfig":
+        try:
+            data = json.loads(content)
+        except json.JSONDecodeError as e:
+            raise PersistError("Error parsing PathPairsConfig: {}".format(str(e)))
```

Ruff (0.15.4)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)